src/Components/Shared/src/DefaultJSDataStream.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components
+{
+    internal sealed class DefaultJSDataStream : BaseJSDataStream
+    {
+        private readonly JSRuntime _runtime;
+        private readonly IJSStreamReference _jsStreamReference;
+        private readonly CancellationToken _streamCancellationToken;
+        private long _offset;
+
+        internal static DefaultJSDataStream CreateJSDataStream(
```

Comment:
The type itself is `internal` anyway

```suggestion
        public static DefaultJSDataStream CreateJSDataStream(
```